irohad/main/application.cpp

```diff
@@ -254,40 +262,76 @@ Irohad::RunResult Irohad::initStorage(
+        [this](std::shared_ptr<shared_model::interface::Block const> block) {
+          iroha::getSubscription()->notify(EventTypes::kOnBlock, block);
+          if (ordering_init and tx_processor and pending_txs_storage_
+              and mst_storage) {
+            ordering_init->processCommittedBlock(block);
+            tx_processor->processCommit(block);
+            for (auto const &completed_tx : block->transactions()) {
+              pending_txs_storage_->removeTransaction(completed_tx.hash());
+              mst_storage->processFinalizedTransaction(completed_tx.hash());
+            }
+            for (auto const &rejected_tx_hash :
+                 block->rejected_transactions_hashes()) {
+              pending_txs_storage_->removeTransaction(rejected_tx_hash);
+              mst_storage->processFinalizedTransaction(rejected_tx_hash);
+            }
           }
-        }
-      };
+        };
+
+    auto st = type == StorageType::kPostgres
+        ? ::iroha::initStorage(*pg_opt_,
+                               pool_wrapper_,
+                               pending_txs_storage_,
+                               query_response_factory_,
+                               config_.block_store_path,
+                               vm_caller_ref,
+                               process_block,
+                               log_manager_->getChild("Storage"))
+        : type == StorageType::kRocksDb
+            ? ::iroha::initStorage(db_context_,
+                                   pending_txs_storage_,
+                                   query_response_factory_,
+                                   config_.block_store_path,
+                                   vm_caller_ref,
+                                   process_block,
+                                   log_manager_->getChild("Storage"))
+            : iroha::expected::makeError("Unexpected storage type.");
+
+    return st | [&](auto &&v) -> RunResult {
+      storage = std::move(v);
+
+      log_->info("[Init] => storage");
+      return {};
+    };
+  };
 
-  IROHA_EXPECTED_TRY_GET_VALUE(
-      storage_,
-      ::iroha::initStorage(*pg_opt_,
-                           pool_wrapper_,
-                           pending_txs_storage_,
-                           query_response_factory_,
-                           config_.block_store_path,
-                           vm_caller_ref,
-                           process_block,
-                           log_manager_->getChild("Storage")));
-
-  storage = std::move(storage_);
-
-  log_->info("[Init] => storage");
-  return {};
+  switch (type) {
+    case StorageType::kPostgres: {
+      IROHA_EXPECTED_TRY_GET_VALUE(
+          pool_wrapper,
+          PgConnectionInit::init(
+              startup_wsv_data_policy, *pg_opt_, log_manager_));
+      pool_wrapper_ = std::move(pool_wrapper);
+    } break;
+
+    case StorageType::kRocksDb: {
+      IROHA_EXPECTED_TRY_GET_VALUE(
+          rdb_port,
+          RdbConnectionInit::init(
```

Comment:
`RdbConnectionInit::init(StartupWsvDataPolicy::kReuse,....)` should not create a folder on filesystem.

Reply:
I don't understand

Reply:
When start iroha with `StartupWsvDataPolicy::kReuse` option. Rocksdb creates a directory on filesystem. Why? It should not.

Reply:
Why? If you run with a parameter ```reuse``` but there is no WSV, then it should!

Reply:
discussed!